### PR TITLE
Implement support for cpu ctrlr and 0-CPU jobs

### PR DESCRIPTION
**Pull request by alexis-cousein**
#### Describe Bug or Feature

See https://pbspro.atlassian.net/wiki/spaces/PD/pages/1627750401/Support+for+the+cpu+cgroup+controller+and+zero-CPU+jobs

#### Describe Your Change

Discussed in the document

#### Link to Design Doc

https://pbspro.atlassian.net/wiki/spaces/PD/pages/1627750401/Support+for+the+cpu+cgroup+controller+and+zero-CPU+jobs

#### Attach Test and Valgrind Logs/Output

N/A -- MoM hook, not one of the main daemons

image

### CODE REVIEW

File: src/hooks/cgroups/pbs_cgroups.PY [hunk 1]
```diff
@@ -4451,7 +4501,40 @@ class CgroupUtils(object):
...
+                    pbs.logmsg(pbs.EVENT_DEBUG,
+                               '%s: Failed to delete cpuset %s'
+                               ' for zero cpus job %s'
+                               % (caller_name(), path, jobid))
+                    pbs.event().reject('Failed to delete cpuset'
+                                       ' for zero cpus job')
+                    # Hopefully the orphan cleaner will eventually
+                    # get rid of it
+            else:
+                # For everything else just pass it down
```
Comment (mike0042) [inline src/hooks/cgroups/pbs_cgroups.PY]: Either remove the else or add a call to pass here.

Comment (alexis-cousein) [reply src/hooks/cgroups/pbs_cgroups.PY]: In my original copy the self.set_limit behind this was indented under the else...let me check the logic.

Comment (alexis-cousein) [reply src/hooks/cgroups/pbs_cgroups.PY]: Yup, somehow an indentation was lost in translation when some things were merged. Go figure.

The logic is that seld.setlimit needs _NOT_ to be done for the two special cases. For "ncpus" there is a set_limit that takes care of multiplying what we need to set by the number of threads if necessarey; for cpuset.cpus we need not call set_limit for ncpus==0 (in that case hostresc['cpuset.cpus'] is empty, and we test for that), but delete the cpuset (after which the processes will be added to the main pbspro.slice cpuset).

In all *other* cases, _then_ we "just" have to call setlimit.

**Comment by mike0042** [comment]
Tests ran as expected. Please re-review @bayucan.

**Pull request merged**